Framework/Kernel/src/AttenuationProfile.cpp

```diff
@@ -15,22 +15,20 @@
 namespace Mantid {
 namespace Kernel {
 
-namespace {
-constexpr auto LARGE_LAMBDA = 100; // Lambda likely to be beyond max lambda in
-                                   // any measured spectra. In Angstroms
-}
-
 /**
  * Construct an attenuation profile object
  * @param inputFileName :: The name of the file containing the attenuation
  * profile data. Filename can be a full path or just file name
  * @param searchPath :: Path to search for the input file
+ * @param extrapolationMaxX :: X value at which a point will be inserted
+ * from the tabulated attenuation data to assist with extrapolation
  * @param extrapolationMaterial :: Material whose properties will be used to
  * extrapolate beyond the lambda range of the supplied profile
  */
 AttenuationProfile::AttenuationProfile(const std::string &inputFileName,
                                        const std::string &searchPath,
-                                       Material *extrapolationMaterial) {
+                                       Material *extrapolationMaterial,
+                                       double extrapolationMaxX) {
```

Comment:
why have we switched from lambda to x?

Reply:
Because the value is  not always the wavelength and can be energy so a neutral variable was used

Reply:
ah ok. If I am being fussy energy can be measured in wavelength (E = hbar/lambda), but lets not get into the pain of hbar in not SI units